src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs

```diff
@@ -30,18 +32,28 @@ internal class CSharpAsAndNullCheckDiagnosticAnalyzer : AbstractCodeStyleDiagnos
 
         public CSharpAsAndNullCheckDiagnosticAnalyzer()
             : base(IDEDiagnosticIds.InlineAsTypeCheckId,
-                   new LocalizableResourceString(
-                       nameof(FeaturesResources.Use_pattern_matching), FeaturesResources.ResourceManager, typeof(FeaturesResources)))
+                new LocalizableResourceString(
+                    nameof(FeaturesResources.Use_pattern_matching), FeaturesResources.ResourceManager, typeof(FeaturesResources)))
         {
         }
 
         protected override void InitializeWorker(AnalysisContext context)
-            => context.RegisterSyntaxNodeAction(SyntaxNodeAction, SyntaxKind.IfStatement);
+            => context.RegisterSyntaxNodeAction(SyntaxNodeAction,
```

Comment:
@mavasani RegisterSyntaxNodeAction could take a SyntaxTree predicate that we could potentially use to avoid calling this action if the SyntaxTree didn't match.